### DISCUSSION TITLE
GuestAgent enabled + IP

### DISCUSSION THREAD

**Discussion by xguitoux** [dougiteixeira/proxmoxve#140] [Feature request] [answered]
Hello, is it possible to have sensors showing if GuestAgent is enabled and the associated IPs to the vm ?

**Comment by dougiteixeira** [comment, marked as answer]
I didn't find this available in the Proxmox API, so it's not possible to add it.

**Comment by n3tkiller** [comment]
@dougiteixeira what about https://pve.proxmox.com/pve-docs/api-viewer/#/nodes/{node}/lxc/{vmid}/interfaces ?

**Comment by mZ738** [comment]
5.2.0 adds both: an **IP address** sensor per VM (through the guest agent) and per container (from `lxc/{vmid}/interfaces` — thanks for the pointer, n3tkiller), with every address by interface as attributes, and a **Guest agent** binary sensor per VM that has the agent configured, on while it answers. Both diagnostic. Reading the VM side needs `VM.GuestAgent.Audit` (`VM.Monitor` on Proxmox 8); a refused read raises a repair that says so. https://github.com/dougiteixeira/proxmoxve/releases/tag/5.2.0
